internal/corazawaf/transaction.go

```diff
@@ -752,6 +752,51 @@ func (tx *Transaction) RequestBodyWriter() io.Writer {
 	return tx.RequestBodyBuffer
 }
 
+type Lenger interface {
+	Len() int
+}
+
+// ProcessRequestBodySize sets the maximum number of bytes that can be read from the
+// request body.
+func (tx *Transaction) AppendRequestBody(r io.Reader) (*types.Interruption, int, error) {
+	if tx.RuleEngine == types.RuleEngineOff {
+		return nil, 0, nil
+	}
+
+	if l, ok := r.(Lenger); ok {
+		if int64(l.Len()) >= tx.RequestBodyLimit {
+			tx.variables.inboundErrorData.Set("1")
+			if tx.WAF.RequestBodyLimitAction == types.RequestBodyLimitActionReject {
+				// We interrupt this transaction in case RequestBodyLimitAction is Reject
+				tx.interruption = &types.Interruption{
+					Status: 403,
+					Action: "deny",
+				}
+				return tx.interruption, 0, nil
+			}
+		}
+	}
+
+	w, err := io.CopyN(tx.RequestBodyBuffer, r, tx.RequestBodyLimit)
+	if err != nil && err != io.EOF {
+		return nil, 0, err
+	}
+
+	if w == tx.RequestBodyLimit {
+		tx.variables.inboundErrorData.Set("1")
+		if tx.WAF.RequestBodyLimitAction == types.RequestBodyLimitActionReject {
```

Comment:
We should handle `RequestBodyLimitActionProcessPartial` here. We might want to implement https://github.com/corazawaf/coraza-proxy-wasm/pull/128#discussion_r1062070317 first - it seems we want to make process methods idempotent